evil-search.el

```diff
@@ -1289,6 +1290,30 @@ a :substitute command with arguments."
   (evil-ex-delete-hl 'evil-ex-substitute)
   (evil-ex-delete-hl 'evil-ex-search))
 
+;; Yank text at point.
+(defvar evil-ex-search-yank-point nil)
+
+(defun evil-search-yank-word (&optional arg)
+  "Pull next word from buffer into search string."
+  (interactive)
+  (let ((fwd-fn #'forward-word)
+        word)
+    (with-current-buffer evil-ex-current-buffer
+      ;; Start to initial point if C-w have never been hit.
+      (unless evil-ex-search-yank-point
+        (setq evil-ex-search-yank-point evil-ex-search-start-point))
```

Comment:
I tried this branch today, and I don't think `evil-ex-search-start-point` is what we should use as a starting value for `evil-ex-search-yank-point`.

With a buffer containing

```
foo barbaz
foo bar123
```

If `point` is at the beginning of the buffer and I start to search forward and type `bar C-w`, I would expect the `C-w` to add `baz` to the search string, since this is what `isearch`'s `C-w` does. But right now we add `foo`, since `foo` is what is in front of `evil-ex-search-start-point`.

i'm thinking we should define another global variable, `evil-ex-search-current-match-beg` (distinct from `evil-ex-search-match-beg`, which seems to correspond to the last search that was actually exited), to be set by `evil-ex-search-goto-offset` that we can use here.

Reply:
please review again

Reply:
Thanks for updating. I spotted several other things.

With a file containing

```
abc def ghi jkl
```

Point is at the beginning of the buffer.

- If I use M-x `isearch-forward` and type `def C-w C-w`, the search string is `def ghi jkl`. However, with this patch, if I use `evil-search-forward` with `evil-search-module` set to `evil-search`, the first `C-w` does not change the search string, and the second one consumes the `ghi`, so that the final search string is `def ghi`.

- If I use M-x `isearch-forward` and type `def C-w <backspace> C-w`, the search string is `def ghi`. With this patch, if I do the same search using `evil-search-forward` and the `evil-search` module, the search string is `def gh jkl`. `<backspace>` should reset the `evil-ex-search-yank-point` to something appropriate.

Reply:
should be fixed

Reply:
Looks good, thanks!